experimental/dds/tree2/src/core/tree/visitDelta.ts

```diff
@@ -175,6 +183,13 @@ function firstPass(delta: Delta.MarkList, visitor: DeltaVisitor, config: PassCon
 					markHasMoveOrDelete = true;
 					break;
 				case Delta.MarkType.MoveOut:
+					if (!config.allowRepairDataMoves && mark.isDelete) {
+						// Handled in the second pass
+						visitModify(index, mark, visitor, config);
+						index += mark.count;
+						break;
```

Comment:
Don't we need this as well?
```suggestion
						markHasMoveOrDelete = true;
						break;
```